plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -217,7 +217,7 @@ class CodeWhispererCodeScanManager(val project: Project) {
             val codeScanSessionConfig = CodeScanSessionConfig.create(file, project, scope)
             language = codeScanSessionConfig.getSelectedFile().programmingLanguage()
             if (scope == CodeWhispererConstants.CodeAnalysisScope.FILE &&
-                (language == CodeWhispererUnknownLanguage.INSTANCE || language.toTelemetryType() == CodewhispererLanguage.Plaintext)
+                !AUTO_FILE_SCAN_SUPPORTED_LANGUAGES.contains(language.toTelemetryType())
```

Comment:
can't we use `CodeWhispererProgrammingLanguage.isCodeScanSupported` interface?

Reply:
```
abstract class CodeWhispererProgrammingLanguage {
    abstract val languageId: String
    open val fileCrawler: FileCrawler = NoOpFileCrawler()

    abstract fun toTelemetryType(): CodewhispererLanguage

    open fun isCodeCompletionSupported(): Boolean = false

    open fun isCodeScanSupported(): Boolean = false
.................

```

Reply:
need to create an additional field the...because file scans support levels are different from Project Scan Support.

Reply:
i see, yea feel free to add a new abstract prop in the language interface, I think it's preferred way to do configuration stuff unless it's not what you want or you have strong preference on the current approach as we better have an unified way to gate all these features which are configured at per language level.

Reply:
Done. updated with new prop.